collect.py

```diff
@@ -92,22 +93,43 @@ def do_something(driver, elem_id=None):
         random.shuffle(children)
 
         for child in children:
-            elem_id = child.get_attribute('id')
 
-            # We need to store the ID in order to replicate what we are doing, so we
-            # have to skip elements with no ID.
-            if elem_id == '':
-                continue
+            # Get all the attributes of the child.
+            child_attributes = driver.execute_script('var elem_attribute = {}; for (i = 0; i < arguments[0].attributes.length; i++) { elem_attribute[arguments[0].attributes[i].name] = arguments[0].attributes[i].value }; return elem_attribute;', child)
 
             # If the element is not displayed or is disabled, the user can't interact with it. Skip
             # non-displayed/disabled elements, since we're trying to mimic a real user.
             if not child.is_displayed() or not child.is_enabled():
                 continue
-
             elem = child
             break
     else:
-        elem = driver.find_element_by_id(elem_id)
+        if 'id' not in elem_attributes.keys():
+            body = driver.find_elements_by_tag_name('body')
+            assert len(body) == 1
+            body = body[0]
+
+            buttons = body.find_elements_by_tag_name('button')
+            links = body.find_elements_by_tag_name('a')
+            inputs = body.find_elements_by_tag_name('input')
+            children = buttons + links + inputs
+
+            for child in children:
+
+                # Get all the attributes of the child.
+                child_attributes = driver.execute_script('var elem_attribute = {}; for (i = 0; i < arguments[0].attributes.length; i++) { elem_attribute[arguments[0].attributes[i].name] = arguments[0].attributes[i].value }; return elem_attribute;', child)
+
+                # If the element is not displayed or is disabled, the user can't interact with it. Skip
```

Comment:
I think we can ignore this part, as we already know that the element was displayed and not enabled in the first browser.

Reply:
@marco-c  Because of two browsers can display in different way it might be that one browser displays the element while other doesn't. Should I remove it?

Reply:
If that happens, it's a compatibility issue and we want to detect it, so it's useful to take a screenshot even if the element is not displayed in the second browser.

Reply:
Oh ok! I get it.